src/state/hooks.ts

```diff
@@ -261,16 +261,6 @@ export const useGetMinBetAmount = () => {
   return useMemo(() => new BigNumber(minBetAmount), [minBetAmount])
 }
 
-export const useGetUserBetByRound = (id: string, account: string): Bet => {
-  const round = useGetRound(id)
-
-  if (!account) {
-    return undefined
-  }
-
-  return round.bets.find((bet) => bet.user.address.toLowerCase() === account.toLocaleLowerCase())
-}
-
```

Comment:
This was the core of the problem. It was looking in the round data which could only contain a max number of 1,000 bets. Replaced by `useGetBetByRoundId`